src/Schema/AST/ASTBuilder.php

```diff
@@ -96,11 +97,15 @@ public function build(): DocumentAST
     protected function applyTypeDefinitionManipulators(): void
     {
         foreach ($this->documentAST->types as $typeDefinition) {
-            foreach (
-                $this->directiveLocator->associatedOfType($typeDefinition, TypeManipulator::class)
-                as $typeManipulator
-            ) {
+            /** @var array<int, bool> $executedManipulators */
+            $executedManipulators = [];
+            while (
+                $typeManipulator = $this->directiveLocator->associatedOfType($typeDefinition, TypeManipulator::class)
+                    ->filter(fn (BaseDirective $typeManipulator) => !isset($executedManipulators[spl_object_id($typeManipulator->directiveNode)]))
```

Comment:
```suggestion
                    ->reject(fn (TypeManipulator $typeManipulator): bool => isset($executedManipulators[spl_object_id($typeManipulator->directiveNode)]))
```

Reply:
Unfortunately, that will result in:
```
Access to an undefined property Nuwave\Lighthouse\Support\Contracts\TypeManipulator::$directiveNode
```

Reply:
I see. Not every directive has to be of type `BaseDirective` though. `associatedOfType` could return a class that just `implements TypeManipulor, Directive`.

In case we do not have access to the directive node, I am not sure what we could use to know if the directive has executed.